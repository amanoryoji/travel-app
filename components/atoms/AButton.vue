<template>
  <button>
    <slot></slot>
  </button>
</template>

<script lang="ts" setup>
import { defineProps } from "vue";
type Props = {
  status: String;
};
const props = defineProps<Props>();
</script>
