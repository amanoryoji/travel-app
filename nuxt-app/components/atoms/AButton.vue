<template>
  <button>{{ props.status }}</button>
</template>

<script lang="ts" setup>
import { defineProps, reactive } from 'vue';
type Props = {
  status: String;
};
const props = defineProps<Props>();
</script>
