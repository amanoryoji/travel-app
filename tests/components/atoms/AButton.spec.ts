import { VueWrapper, shallowMount } from "@vue/test-utils";

import AButton from "/components/atoms/AButton.vue";


describe("AButton", () => {
  let wrapper: VueWrapper<AButton>;
  beforeEach(() => {
    wrapper = shallowMount(AButton, {
      propsData: {
        label: "ボタン名",
      },
    });
  });

  it("コンポーネントがマウントされる", () => {
    expect(wrapper.vm).toBeTruthy();
  });

  it("labelの名前がボタン名になっている", () => {
    expect(wrapper.props().label).toBe('ボタン名')
  });

  describe("スナップショット", () => {
    it("必須Propのみ", () => {
      expect(wrapper.html()).toMatchSnapshot();
    });

    it("全Prop指定", async () => {
      await wrapper.setProps({
        variant: "seconcary",
        size: "lg",
        pill: true,
        block: true,
        full: true,
        disabled: true,
      });
      expect(wrapper.html()).toMatchSnapshot();
    });

    describe("variant", () => {
      it("primary", async () => {
        await wrapper.setProps({ size: "primary" });
        expect(wrapper.html()).toMatchSnapshot();
      });
      it("secondary", async () => {
        await wrapper.setProps({ size: "secondary" });
        expect(wrapper.html()).toMatchSnapshot();
      });
      it("tertiary", async () => {
        await wrapper.setProps({ size: "tertiary" });
        expect(wrapper.html()).toMatchSnapshot();
      });
      it("outline-primary", async () => {
        await wrapper.setProps({ size: "outline-primary" });
        expect(wrapper.html()).toMatchSnapshot();
      });
      it("outline-secondary", async () => {
        await wrapper.setProps({ size: "outline-secondary" });
        expect(wrapper.html()).toMatchSnapshot();
      });
      it("outline-tertiary", async () => {
        await wrapper.setProps({ size: "outline-tertiary" });
        expect(wrapper.html()).toMatchSnapshot();
      });
    });

    describe("size", () => {
      it("sm", async () => {
        await wrapper.setProps({ size: "sm" });
        expect(wrapper.html()).toMatchSnapshot();
      });

      it("md", async () => {
        await wrapper.setProps({ size: "md" });
        expect(wrapper.html()).toMatchSnapshot();
      });

      it("lg", async () => {
        await wrapper.setProps({ size: "lg" });
        expect(wrapper.html()).toMatchSnapshot();
      });
    });

    describe("icon", () => {
      it("not found", async () => {
        await wrapper.setProps({ icon: "/path_to_image" });
        expect(wrapper.html()).toMatchSnapshot();
      });

      it("account_circle", async () => {
        await wrapper.setProps({ icon: "/images/icon/account_circle.svg" });
        expect(wrapper.html()).toMatchSnapshot();
      });
    });

    describe("isIconRight", () => {
      it("false", async () => {
        await wrapper.setProps({
          icon: "/images/icon/account_circle.svg",
          isIconRight: false,
        });
        expect(wrapper.html()).toMatchSnapshot();
      });

      it("true", async () => {
        await wrapper.setProps({
          icon: "/images/icon/account_circle.svg",
          isIconRight: true,
        });
        expect(wrapper.html()).toMatchSnapshot();
      });
    });
  });
});
