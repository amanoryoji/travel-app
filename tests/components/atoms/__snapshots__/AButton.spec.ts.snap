// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AButton > スナップショット > icon > account_circle 1`] = `
"<button data-v-70506b90=\\"\\" type=\\"button\\" class=\\"bl_button bl_button__primary bl_button__md\\"><img data-v-70506b90=\\"\\" class=\\"bl_button_icon\\" src=\\"/images/icon/account_circle.svg\\" alt=\\"\\"><span data-v-70506b90=\\"\\" class=\\"bl_button_label\\">ボタン名</span>
  <!--v-if-->
  <!-- <div v-show=\\"loading\\" class=\\"bl_button_loadingSpinner\\">
      <LoadingSpinner size=\\"xxsm\\" />
    </div> -->
</button>"
`;

exports[`AButton > スナップショット > icon > not found 1`] = `
"<button data-v-70506b90=\\"\\" type=\\"button\\" class=\\"bl_button bl_button__primary bl_button__md\\"><img data-v-70506b90=\\"\\" class=\\"bl_button_icon\\" src=\\"/path_to_image\\" alt=\\"\\"><span data-v-70506b90=\\"\\" class=\\"bl_button_label\\">ボタン名</span>
  <!--v-if-->
  <!-- <div v-show=\\"loading\\" class=\\"bl_button_loadingSpinner\\">
      <LoadingSpinner size=\\"xxsm\\" />
    </div> -->
</button>"
`;

exports[`AButton > スナップショット > isIconRight > false 1`] = `
"<button data-v-70506b90=\\"\\" type=\\"button\\" class=\\"bl_button bl_button__primary bl_button__md\\"><img data-v-70506b90=\\"\\" class=\\"bl_button_icon\\" src=\\"/images/icon/account_circle.svg\\" alt=\\"\\"><span data-v-70506b90=\\"\\" class=\\"bl_button_label\\">ボタン名</span>
  <!--v-if-->
  <!-- <div v-show=\\"loading\\" class=\\"bl_button_loadingSpinner\\">
      <LoadingSpinner size=\\"xxsm\\" />
    </div> -->
</button>"
`;

exports[`AButton > スナップショット > isIconRight > true 1`] = `
"<button data-v-70506b90=\\"\\" type=\\"button\\" class=\\"bl_button bl_button__primary bl_button__md\\">
  <!--v-if--><span data-v-70506b90=\\"\\" class=\\"bl_button_label\\">ボタン名</span><img data-v-70506b90=\\"\\" src=\\"/images/icon/account_circle.svg\\" alt=\\"\\">
  <!-- <div v-show=\\"loading\\" class=\\"bl_button_loadingSpinner\\">
      <LoadingSpinner size=\\"xxsm\\" />
    </div> -->
</button>"
`;

exports[`AButton > スナップショット > size > lg 1`] = `
"<button data-v-70506b90=\\"\\" type=\\"button\\" class=\\"bl_button bl_button__primary bl_button__lg\\">
  <!--v-if--><span data-v-70506b90=\\"\\" class=\\"bl_button_label\\">ボタン名</span>
  <!--v-if-->
  <!-- <div v-show=\\"loading\\" class=\\"bl_button_loadingSpinner\\">
      <LoadingSpinner size=\\"xxsm\\" />
    </div> -->
</button>"
`;

exports[`AButton > スナップショット > size > md 1`] = `
"<button data-v-70506b90=\\"\\" type=\\"button\\" class=\\"bl_button bl_button__primary bl_button__md\\">
  <!--v-if--><span data-v-70506b90=\\"\\" class=\\"bl_button_label\\">ボタン名</span>
  <!--v-if-->
  <!-- <div v-show=\\"loading\\" class=\\"bl_button_loadingSpinner\\">
      <LoadingSpinner size=\\"xxsm\\" />
    </div> -->
</button>"
`;

exports[`AButton > スナップショット > size > sm 1`] = `
"<button data-v-70506b90=\\"\\" type=\\"button\\" class=\\"bl_button bl_button__primary bl_button__sm\\">
  <!--v-if--><span data-v-70506b90=\\"\\" class=\\"bl_button_label\\">ボタン名</span>
  <!--v-if-->
  <!-- <div v-show=\\"loading\\" class=\\"bl_button_loadingSpinner\\">
      <LoadingSpinner size=\\"xxsm\\" />
    </div> -->
</button>"
`;

exports[`AButton > スナップショット > variant > outline-primary 1`] = `
"<button data-v-70506b90=\\"\\" type=\\"button\\" class=\\"bl_button bl_button__primary bl_button__outline-primary\\">
  <!--v-if--><span data-v-70506b90=\\"\\" class=\\"bl_button_label\\">ボタン名</span>
  <!--v-if-->
  <!-- <div v-show=\\"loading\\" class=\\"bl_button_loadingSpinner\\">
      <LoadingSpinner size=\\"xxsm\\" />
    </div> -->
</button>"
`;

exports[`AButton > スナップショット > variant > outline-secondary 1`] = `
"<button data-v-70506b90=\\"\\" type=\\"button\\" class=\\"bl_button bl_button__primary bl_button__outline-secondary\\">
  <!--v-if--><span data-v-70506b90=\\"\\" class=\\"bl_button_label\\">ボタン名</span>
  <!--v-if-->
  <!-- <div v-show=\\"loading\\" class=\\"bl_button_loadingSpinner\\">
      <LoadingSpinner size=\\"xxsm\\" />
    </div> -->
</button>"
`;

exports[`AButton > スナップショット > variant > outline-tertiary 1`] = `
"<button data-v-70506b90=\\"\\" type=\\"button\\" class=\\"bl_button bl_button__primary bl_button__outline-tertiary\\">
  <!--v-if--><span data-v-70506b90=\\"\\" class=\\"bl_button_label\\">ボタン名</span>
  <!--v-if-->
  <!-- <div v-show=\\"loading\\" class=\\"bl_button_loadingSpinner\\">
      <LoadingSpinner size=\\"xxsm\\" />
    </div> -->
</button>"
`;

exports[`AButton > スナップショット > variant > primary 1`] = `
"<button data-v-70506b90=\\"\\" type=\\"button\\" class=\\"bl_button bl_button__primary bl_button__primary\\">
  <!--v-if--><span data-v-70506b90=\\"\\" class=\\"bl_button_label\\">ボタン名</span>
  <!--v-if-->
  <!-- <div v-show=\\"loading\\" class=\\"bl_button_loadingSpinner\\">
      <LoadingSpinner size=\\"xxsm\\" />
    </div> -->
</button>"
`;

exports[`AButton > スナップショット > variant > secondary 1`] = `
"<button data-v-70506b90=\\"\\" type=\\"button\\" class=\\"bl_button bl_button__primary bl_button__secondary\\">
  <!--v-if--><span data-v-70506b90=\\"\\" class=\\"bl_button_label\\">ボタン名</span>
  <!--v-if-->
  <!-- <div v-show=\\"loading\\" class=\\"bl_button_loadingSpinner\\">
      <LoadingSpinner size=\\"xxsm\\" />
    </div> -->
</button>"
`;

exports[`AButton > スナップショット > variant > tertiary 1`] = `
"<button data-v-70506b90=\\"\\" type=\\"button\\" class=\\"bl_button bl_button__primary bl_button__tertiary\\">
  <!--v-if--><span data-v-70506b90=\\"\\" class=\\"bl_button_label\\">ボタン名</span>
  <!--v-if-->
  <!-- <div v-show=\\"loading\\" class=\\"bl_button_loadingSpinner\\">
      <LoadingSpinner size=\\"xxsm\\" />
    </div> -->
</button>"
`;

exports[`AButton > スナップショット > 全Prop指定 1`] = `
"<button data-v-70506b90=\\"\\" type=\\"button\\" class=\\"bl_button bl_button__seconcary bl_button__lg bl_button__pill bl_button__block bl_button__full bl_button__disabled\\" disabled=\\"\\">
  <!--v-if--><span data-v-70506b90=\\"\\" class=\\"bl_button_label\\">ボタン名</span>
  <!--v-if-->
  <!-- <div v-show=\\"loading\\" class=\\"bl_button_loadingSpinner\\">
      <LoadingSpinner size=\\"xxsm\\" />
    </div> -->
</button>"
`;

exports[`AButton > スナップショット > 必須Propのみ 1`] = `
"<button data-v-70506b90=\\"\\" type=\\"button\\" class=\\"bl_button bl_button__primary bl_button__md\\">
  <!--v-if--><span data-v-70506b90=\\"\\" class=\\"bl_button_label\\">ボタン名</span>
  <!--v-if-->
  <!-- <div v-show=\\"loading\\" class=\\"bl_button_loadingSpinner\\">
      <LoadingSpinner size=\\"xxsm\\" />
    </div> -->
</button>"
`;
